crmd/lrm_state.c

```diff
@@ -222,6 +229,11 @@ lrm_state_reset_tables(lrm_state_t * lrm_state)
                   g_hash_table_size(lrm_state->rsc_info_cache));
         g_hash_table_remove_all(lrm_state->rsc_info_cache);
     }
+    if (lrm_state->metadata_cache) {
+        crm_trace("Re-setting metadata cache with %d members",
+                  g_hash_table_size(lrm_state->metadata_cache));
+        g_hash_table_remove_all(lrm_state->metadata_cache);
+    }
```

Comment:
Thinking about this some more, I think we want to reset the metadata cache for the remote node when its connection *starts*.

If the remote connection migrates to another node, the original connection host may still need to call build_operation_update() for operations that ran while it was the host. Its cache entries will still be valid for this purpose, so we don't want to clear it on migration. However, if a resource starts on the remote node after the connection has migrated, the metadata will be updated on the new host, not the original host; if the connection later migrates back to the original host, it could have stale metadata at that point. So, we can reset the cache at that point.